features/images.feature

```diff
@@ -0,0 +1,40 @@
+@images
+Feature: OpenStack Image management
+  As an administrator
+  I want to make sure that OpenStack Image management component
+  is working correctly
+
+  @create_kvm
+  Scenario: Creation of KVM image
+    Given KVM source image exists
+    And glance image does not exist
+    When I create new glance image based on jeos
+    Then this image has non-empty ID
+    And its ID can be used to show the image info
+    And the status of the image is active
+    When I delete the image
```

Comment:
> But the discussion you are pointing to surely doesn't show any agreement if we should do it in the After hooks or not.

We didn't reach agreement yet, but yes I am proposing to use both backgrounds and an `After` hook.

> In general, I still find current approach more readable and understandable, although yours might be more pure and principally better.

Well some part of it is personal taste, yes.  Personally I find the combination of this with an `After` hook for cleanup more readable and understandable:

```gherkin
Background:
  Given glance image does not exist
  And image quota has at least 2GB remaining

Scenario: Creation of XEN HVM image
  Given XEN HVM source image exists
  When I create new glance image based on jeos
  Then this image has non-empty ID
  And its ID can be used to show the image info
  And the status of the image is active
```

because:

- it does not violate the DRY rule by repeating the same 3 steps in every scenario
- each scenario has 5 steps instead of 8, so is easier to read
- each scenario just tests a single thing
- setup and teardown are cleanly separated from the actual test
- the test harness can avoid spamming the report with setup and teardown (which you should never have to care about unless it fails)
- the test harness can more clearly distinguish between failures in tests and failures in setup/teardown
- even if previous teardown failed (e.g. due to a bug in image deletion), it will check that there is still sufficient quota to create the new image

I'm not sure why you would prefer to continue breaking the DRY rule by not using backgrounds or an `After` hook, but even if you do, it would be better to define a nested step e.g.

```ruby
And /^the image can be deleted$/ do
  steps %{
    When I delete the image
    Then it is no longer listed
  }
end
```

so that at least you are only duplicating 2 lines per scenario instead of 3.

BTW this long conversation about this one PR may feel like pointless bike-shedding, but I think it's actually really worthwhile because `cct` is in its very early days, so any feature-writing style which is established now will certainly be copied by other people as they join in the effort to write new features and steps.  So I think we should try to set a really good precedent for others to follow.